src/main/java/WordInMatrix.java

```diff
@@ -0,0 +1,133 @@
+            letterPosition = letterPosition - row;
+            cellsSequence.add(addToCellsSequence(letterPosition, row));
+            matrix = markMyWay(matrix, letterPosition, word.charAt(0));
+            cellsSequence = checkSells(matrix, word.substring(1),
+                    letterPosition, row, cellsSequence);
+            if (cellsSequence.get(cellsSequence.size() - 1).equals("Wrong word")) {
+                cellsSequence.remove("Wrong word");
+                cellsSequence.remove(cellsSequence.size() - 1);
+                letterPosition += row;
+            } else {
+                return cellsSequence;
+            }
+        }
+        if (letterPosition + row < matrix.length()
+                && word.charAt(0) == matrix.charAt(letterPosition + row)) {
+            letterPosition = letterPosition + row;
+            cellsSequence.add(addToCellsSequence(letterPosition, row));
+            matrix = markMyWay(matrix, letterPosition, word.charAt(0));
+            cellsSequence = checkSells(matrix, word.substring(1),
+                    letterPosition, row, cellsSequence);
+            if ((cellsSequence.get(cellsSequence.size() - 1).equals("Wrong word"))) {
+                cellsSequence.remove("Wrong word");
+                cellsSequence.remove(cellsSequence.size() - 1);
+                letterPosition -= row;
+            } else {
+                return cellsSequence;
+            }
+
+        }
+        if (letterPosition % row != 0 && word.charAt(0) == matrix.charAt(letterPosition - 1)) {
+            letterPosition = letterPosition - 1;
+            cellsSequence.add(addToCellsSequence(letterPosition, row));
+            matrix = markMyWay(matrix, letterPosition, word.charAt(0));
+            cellsSequence = checkSells(matrix, word.substring(1),
+                    letterPosition, row, cellsSequence);
+            if (cellsSequence.get(cellsSequence.size() - 1).equals("Wrong word")) {
+                cellsSequence.remove("Wrong word");
+                cellsSequence.remove(cellsSequence.size() - 1);
+                letterPosition += 1;
+            } else {
+                return cellsSequence;
+            }
+
+        }
+        if (letterPosition % row != row - 1
+                && word.charAt(0) == matrix.charAt(letterPosition + 1)) {
+            letterPosition = letterPosition + 1;
+            cellsSequence.add(addToCellsSequence(letterPosition, row));
+            matrix = markMyWay(matrix, letterPosition, word.charAt(0));
+            cellsSequence = checkSells(matrix, word.substring(1),
+                    letterPosition, row, cellsSequence);
+            return cellsSequence;
+        }
+        cellsSequence.add("Wrong word");
+        return cellsSequence;
+    }
+
+    private String markMyWay(String matrix, int letterPosition, char letterToChange) {
+        StringBuilder changeString = new StringBuilder(matrix);
+        changeString.setCharAt(letterPosition, (char) (letterToChange + 32));
```

Comment:
it's to declare 32 as a constant, it's not quite clear why it's 32 and not 33 etc